contracts/modules/claims/Incidents.sol

```diff
@@ -0,0 +1,234 @@
+/* Copyright (C) 2021 NexusMutual.io
+
+  This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+    along with this program.  If not, see http://www.gnu.org/licenses/ */
+
+pragma solidity ^0.5.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "../../abstract/MasterAware.sol";
+import "../../interfaces/IPooledStaking.sol";
+import "../capital/Pool.sol";
+import "../claims/ClaimsData.sol";
+import "../claims/ClaimsReward.sol";
+import "../cover/QuotationData.sol";
+import "../governance/MemberRoles.sol";
+import "../token/TokenController.sol";
+
+contract Incidents is MasterAware {
+  using SafeERC20 for IERC20;
+  using SafeMath for uint;
+
+  struct Incident {
+    uint32 date;
+    uint112 priceBefore;
+    uint112 priceAfter;
+    address coveredToken;
+  }
+
+  // contract identifiers
+  enum ID {CD, CR, QD, TC, MR, P1, PS}
+
+  mapping(uint => address payable) contracts;
+
+  Incident[] public incidents;
+
+  // covered token => peg token
+  mapping(address => address) public underlyingTokens;
+
+  // claim id => payout amount
+  mapping(uint => uint) public claimPayouts;
+
+  function addUnderlyingToken(
+    address coveredToken,
+    address underlyingToken
+  ) external onlyGovernance {
+    require(underlyingTokens[coveredToken] == address(0), "Incidents: Underlying token already set");
+    underlyingTokens[coveredToken] = underlyingToken;
+  }
+
+  function addIncident(
+    address coveredToken,
+    uint incidentDate,
+    uint priceBefore,
+    uint priceAfter
+  ) external onlyGovernance {
+
+    require(priceBefore > priceAfter, "Incidents: No depeg");
```

Comment:
this condition can be stronger since we have 10% minimum price de-peg in the cover wording. however if we put it here we'd need to do a contract upgrade if we change that.

Reply:
This is subjective and there was a discussion whether to include this check in the code or not. Let's discuss this with the team.